### PR TITLE
Hotfix/test fixtures

### LINKED ISSUE #315: Test fixtures
Fixtures for tests in punchcard have been hastily slapdashed in a dogpile of nope. We need them to use a more common directory structure and old/outdated files removed.

```gherkin
Feature: Test Fixtures cleanup
  As a core developer
  I want a clean, easy-to-understand directory containing my tests fixtures
  So that I can have a quick way to mimic Punchcard and content-types output.

  Scenario: Common directory
    Given that fixtures currently reside in a haphazard set of folders under `/tests`
      And all fixtures should be under the `/tests/fixtures` directory
     When traversing through Punchcards tests directory
     Then all fixtures should be under the `/tests/fixtures` directory

  Scenario: New common fixtures
    Given that I need common fixtures across test files
    When updating or creating new tests
    Then I should have some common fixtures which represent the needs of various Punchcard functions
      And there should be a fixture object for multiple content types after yaml import, after merging with input plugins
      And there should be a fixture object for multiple workflows after yaml import.
```

---

**Test fixtures**
- [x] Common Directory
- [x] New common fixtures

### PR DESCRIPTION
**Pull request by scottnath**
Fixtures have been re-arranged and now the tests properly point to the actual test fixtures.

---
Resolves #315

`DCO 1.1 Signed-off-by: Scott Nath <github@scottnath.com>`

### CODE REVIEW

File: config/default-test.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 'use strict';
 
+const _ = require('lodash');
 const path = require('path');
 
 const defaultCfg = require('./default');
 
+const config = _.cloneDeep(defaultCfg);
```
Comment (Snugug) [inline config/default-test.js]: Why are you cloning this? The `config` module will deep merge these together, so we should only need to override the specific object we want to change

File: lib/workflows/load.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ const load = (wfpath) => {
     dir.readFiles(workflowsPath, {
       match: /.yml$/,
       exclude: /^\./,
+      recursive: false,
```
Comment (Snugug) [inline lib/workflows/load.js]: Why are you turning off recursive find?

File: package.json [hunk 3]
```diff
@@ -20,8 +20,8 @@
     "posttest": "gulp build",
     "start": "nodemon -e html,js ./index.js",
     "lint": "gulp lint",
-    "ava:watch": "NODE_CONFIG_DIR=../config ava --watch | tap-diff",
-    "ava": "NODE_CONFIG_DIR=../config ava | tap-diff",
+    "ava:watch": "NODE_ENV=default-test NODE_CONFIG_DIR=../config ava --watch | tap-diff",
```
Comment (Snugug) [inline package.json]: This _should_ just be "test" and maybe we rename the test configuration to go with it

File: tests/workflows.js [hunk 4]
```diff
@@ -380,5 +342,5 @@ test('workflows in type', t => {
   noflow.workflow = '';
   const nopeflow = workflows.workflow(noflow, allFlows, globalConfig, req);
 
-  t.is(nopeflow, false, 'Returns false on workflow missing from global flows');
+  t.is(nopeflow, allFlows[1], 'Returns false on workflow missing from global flows');
```
Comment (Snugug) [inline tests/workflows.js]: Using array index feels brittle. Thoughts?

File: tests/fixtures/content-types/all-types-merged.js [hunk 5]
```diff
@@ -0,0 +1,281 @@
+/**
```
Comment (Snugug) [inline tests/fixtures/content-types/all-types-merged.js]: Either `all-types` or `all-merged` but not `all-types-merged` please.

File: config/default-test.js (hunk 1 above)
Comment (scottnath) [inline config/default-test.js]: I see. So in this case, I need to make a change to `config.content` so I should cloneDeep just that object you're saying?

File: lib/workflows/load.js (hunk 2 above)
Comment (scottnath) [inline lib/workflows/load.js]: will remove

File: package.json (hunk 3 above)
Comment (scottnath) [inline package.json]: agreed....so then `./config/test.js` ..._but_ is that the file which travis will look for?

**Pull request merged**